### ISSUE TITLE
`enum_source_class.dart` contains faulty mixin code in dart 3

### ISSUE THREAD

**Issue by ahmednfwela** [google/built_value.dart#1248]
this file
https://github.com/google/built_value.dart/blob/master/built_value_generator/lib/src/enum_source_class.dart#L184

checks code similar to this

```dart
abstract class AMixin = Object with _$AMixin;
```

which uses the generated code:
```dart
abstract class _$AMixin {
  // ignore: non_constant_identifier_names
  _$AMeta get A => const _$AMeta();
}
```

which in dart 3 causes compile-time error 

> The class '`_$AMixin`' can't be used as a mixin because it's neither a mixin class nor a mixin.dart [class_used_as_mixin](https://dart.dev/diagnostics/class_used_as_mixin)

faulty line: https://github.com/google/built_value.dart/blob/master/built_value_generator/lib/src/enum_source_class.dart#LL247C11-L247C11

should be 
```diff
-      ..writeln('abstract class _\$${name}Mixin {')
+      ..writeln('mixin _\$${name}Mixin {')
```

cc @davidmorgan

**Comment by davidmorgan** [comment]
I'm not sure if we ever announced this feature, it was for use with Angular templates--is that where you're using it?

I didn't fix it because I thought possibly nobody was using it any more :) but if it still does something useful I can take a look. Thanks.

**Comment by ahmednfwela** [comment]
it's used by the "dart-dio" openapi-generator and is generated for all enums in a spec, so yes I think it will be useful 😊

**Comment by davidmorgan** [comment]
Good to know! Any chance you could point to an example, please? That way I can make sure it still works as intended for that use case. Thanks.

**Comment by ahmednfwela** [comment]
for example:
https://github.com/banool/aptos_api_dart/blob/66e1d4a9514b5a50348b129634feab2e87c984a6/lib/src/model/role_type.dart#L32

**Comment by davidmorgan** [comment]
Hi @ahmednfwela 

I think #1253 should do it. Unfortunately, code using built_value will need an update, as the old way `class TestEnumMixin = Object with _$TestEnumMixin` is not valid as a mixin in Dart 3.

Instead I think the right way is `typedef TestEnumMixin = _$TestEnumMixin`, so my PR adds triggering of generation based on this.

What do you think, please?

Thanks.

**Comment by ahmednfwela** [comment]
looks good! thanks for fixing it

**Comment by davidmorgan** [comment]
Released in `8.6.1`. Thanks again for reporting! Please let me know if you hit any further issues.
